### PR TITLE
add interface Menu

### PR DESCRIPTION
**Pull request by gafilianog**

### CODE REVIEW

File: src/com/github/argajuvi/Main.java [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Scanner;
-import java.util.stream.Collectors;
 
 public class Main {
 
-    private final List<Product> productList;
-    private final List<User> userList;
+    public static List<Product> productList = null;
```
Comment (Alviannn) [inline src/com/github/argajuvi/Main.java]: Langsung assign value disini bisa aja, but it's fine lah

File: src/com/github/argajuvi/Main.java [hunk 2]
```diff
@@ -104,11 +102,15 @@ public void loginUser(List<User> users) {
                         // VALIDASI APAKAH LOGIN UNTUK ADMIN ATAU USER
                         if (username.equals("admin")) {
                             // SHOW MENU ADMIN KALAU YANG LOGIN ADMIN
-                            this.showAdminMenu();
+//                            this.showAdminMenu();
+                            Menu adminMenu = new AdminMenu();
```
Comment (Alviannn) [inline src/com/github/argajuvi/Main.java]: Kenapa gk buat variable `Menu menu;` aja di luar `if-else` ini? It's cleaner that way m8

File: src/com/github/argajuvi/menus/ProductMenu.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        Utils.waitForEnter();
+    }
+
+    /**
+     * Shows the list of products in a table view
+     *
+     * @param filterType the type of product should be shown on the view,
+     *                   insert {@code null} to not use any filters
+     */
+    public static void showProductsView(@Nullable Product.Type filterType) {
```
Comment (Alviannn) [inline src/com/github/argajuvi/menus/ProductMenu.java]: I have to say, this isn't the best way to implement it but I don't really mind it.

Menurut gw sih ini bisa buat 1 class misalnya `Views` di package `utils` trus masukin itu semua function yg berhubungan sama _views_, ato kalo mau versi malesnya taro di class `Main`

File: src/com/github/argajuvi/Main.java [hunk 4]
```diff
@@ -1,49 +1,51 @@
...
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class Main {
 
-    private final List<Product> productList;
-    private final List<User> userList;
-
-    private User currentUser = null;
+    public static List<Product> productList = Stream.of(
```
Comment (Alviannn) [inline src/com/github/argajuvi/Main.java]: I just remembered, semua yg `public static` bagusnya dikasih nama upper snake-case, misalnya ini `productList` jadinya `PRODUCT_LIST`. Do you want me to take care of this or nah?

Comment (gafilianog) [reply src/com/github/argajuvi/Main.java]: lemme do it. gw baru tau ttg ini, kirain cuma konstanta

Comment (Alviannn) [reply src/com/github/argajuvi/Main.java]: I mean... technically itu harusnya konstanta, `List` tuh kalo di `final` masih tetap mutable.

**Review by Alviannn — APPROVED** [review]

**Pull request merged**